code/modules/mob/living/simple_animal/bot/mulebot.dm

```diff
@@ -543,13 +544,20 @@
 	B.setDir(direct)
 	bloodiness--
 
-/mob/living/simple_animal/bot/mulebot/Moved() //make sure we always use power after moving.
+/**
+ * Signal handler for COMSIG_MOVABLE_CROSSED_OVER signals sent by this mulebot.
+ *
+ * Intended to be used to crush various things.
+ */
+/mob/living/simple_animal/bot/mulebot/proc/on_crossed_over(atom/movable/source, atom/movable/crossed_atom)
+	SIGNAL_HANDLER
+
+	if(ishuman(crossed_atom))
+		run_over(crossed_atom)
+
+/mob/living/simple_animal/bot/mulebot/Moved()
 	. = ..()
-	if(!cell)
-		return
-	cell.use(cell_move_power_usage)
-	if(cell.charge < cell_move_power_usage) //make sure we have enough power to move again, otherwise turn off.
-		turn_off()
```

Comment:
so just to double check, a player controlled bot with no power cell is unable to move, correct? And a player controlled bot with a low powercell automatically turns off when they run out of power too?

Reply:
Player-controlled bots are CBT and I've added a new set of signals at the client/Move() level which are just about the only way to tell when a given mob has moved because a `client` did it as opposed to being dragged, pushed or whatever else - At least by my knowledge.

Automated bots move via process() and listen for new signals sent from there in order to check power, turn themselves off, block movement etc.
![image](https://user-images.githubusercontent.com/24975989/111355379-3a2e1800-867f-11eb-9791-25eead686538.png)

And drain power in the same manner.
![image](https://user-images.githubusercontent.com/24975989/111355728-8e38fc80-867f-11eb-8eb3-610fcade2d89.png)

Meanwhile player controlled mulebots call the same procs through new signals sent from client/Move() instead.
![image](https://user-images.githubusercontent.com/24975989/111356438-46ff3b80-8680-11eb-85d5-ca9c440d6fe7.png)
![image](https://user-images.githubusercontent.com/24975989/111355891-b7598d00-867f-11eb-9ddc-c4b9f3f9a6d5.png)

The bitflags returned by all these new signals are just aliases of COMPONENT_MOVABLE_BLOCK_PRE_MOVE and are interchangeable as a result.

This allows mulebots to handle all the logic required to decide if they are able to move either due to automated movement in process() or because a client controls them.

![4RWd8xbQ9k](https://user-images.githubusercontent.com/24975989/111358289-14564280-8682-11eb-93b5-0d75dc9fcb55.gif)